src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -122,14 +154,19 @@ private static boolean doSpin(final int numOfSpins) {
      */
     private static boolean doSpin() {
         boolean result;
-        if(state != STATE_NOT_CREATED) {
-            result = dev.spin();
-            appendTrace("spin : performing a spin");
-        } else {
-            appendTrace("doSpin : Spin not valid: State is NOT_CREATED");
+        if (state == STATE_NOT_CREATED) {
+            appendTrace("doSpin: Error, State is NOT_CREATED");
             result = false;
         }
```

Comment:
Again, I think this is nit picky, but to keep with the current convention put the else and the bracket on the same line if it isn't already. Github's formatting shows it on a separate line.

Also, I'm not sure if the isSpinValid method is being used, if it is then state == STATE_NOT_CREATED is already being checked, and then you can move the if (dev == null) into isSpinValid. That would then leave doSpin() to:

```
if(isSpinValid()) {
     // doSpin
} else { 
    result = false
} 
return result
```

Reply:
I have included this implementation in my bugfixes branch that will be commited soon. :)